### PR TITLE
[HttpFoundation] [Request] Case insensitive check on content_type in ServerBag

### PR DESCRIPTION
**Pull request by ghost**
| Q | A |
| --- | --- |
| Branch? | 2.8 |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | comma-separated list of tickets fixed by the PR, if any |
| License | MIT |
| Doc PR | reference to the documentation PR, if any |

ServerBag data extraction\replace could be case insensitive.

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 1]
```diff
@@ -363,7 +363,10 @@ public static function create($uri, $method = 'GET', $parameters = array(), $coo
             case 'POST':
             case 'PUT':
             case 'DELETE':
-                if (!isset($server['CONTENT_TYPE'])) {
+                if (isset($server['content_type']) {
```
Comment (ro0NL) [inline src/Symfony/Component/HttpFoundation/Request.php]: What about normalizing all passed server keys uppercase before merging above..

Comment (Pherserk) [inline src/Symfony/Component/HttpFoundation/Request.php]: I was thinking about it, but this seems the most conservative change.
Probably there is so many software using lowercase keys, that could break.

Comment (ro0NL) [inline src/Symfony/Component/HttpFoundation/Request.php]: Ok.. maybe fix it on the other side then? Ie. make sure the value is passed uppercase, as this comes from header/server conversion right?

Comment (Pherserk) [inline src/Symfony/Component/HttpFoundation/Request.php]: This sound like an integration scenario.
But is always possible to pass anything you want into the $server parameter, e.g. in a functional test.

**Comment by Pherserk** [comment]
making it again in one single commit

File: src/Symfony/Component/HttpFoundation/Request.php (hunk 1 above)
Comment (ro0NL) [inline src/Symfony/Component/HttpFoundation/Request.php]: Agree..  but if it's expects uppercase keys by design it's not really an issue. And passing in a lowercased version somewhere in the core is a bug :)

**Pull request closed without merging**